### ISSUE TITLE
Extracting boundary features from a separate plugin

### ISSUE THREAD

**Issue by harshilpatel312** [jongough/ocpn_draw_pi#279]
Jon,

What would you suggest for extracting the boundary features (points, types and state) from existing boundaries in your plugin?

I'm writing a separate plugin, and I'd love to use the boundaries which already exist in your plugin, but I don't know how to get them. I've looked at ODAPI.h but the functions in it do not return the points, type and state of the boundaries specifically. 

It looks like its possible to access that data via a global variable (g_pBoundaryList), but that seems like a bit of a hack, as well. 

Ideally, I'd like to get a message that just lists these boundary features, but what do you think? 

Harshil

**Comment by jongough** [comment]
Hi Harshil,
   The plugin does not externalise the boundary information. The JSON messages and ODAPI are used to find if points are inside/outside a boundary or if a line crosses a boundary. The plugin returns information about the boundary (boundary areas and boundary points), but this is to do with the type and identifier of the boundary.

   What do you want to do that needs more information about boundaries and what information do you need?

   The ODnavobj.xml file has all the information about objects that have been drawn, but this is used, like navobj.xml in mainline, to handle restarts so that the user does not loose information.

Jon

**Comment by harshilpatel312** [comment]
Jon,

I would like my plugin to keep a track of all the boundaries (and it's features) drawn with ocpn_draw_pi. In case the user updates the boundaries, I would like the changes to be reflected in my plugin. For now, I think the boundary points (lat, lon), boundary type and state should suffice.    

As you mentioned, ODnavobj.xml is used to handle restarts. I am actually looking for something more "real-time".

Best,
Harshil

**Comment by jongough** [comment]
Hi Harshil,
   What are you trying to do that needs real time information about boundaries and boundary points?

   A change like this would have to be in the next version of the plugin so that it can go through the beta process, so it may not happen for a few months.

Jon

**Comment by harshilpatel312** [comment]
Thanks for your reply, Jon!

I'm just developing a plugin that tracks boundary data. I understand its a quite a change but I'm willing to work on it myself, just need some direction regarding it.

Just an update about the extern variable, g_pBoundaryList: I included the variable in my code and was able to compile the code successfully. However, when I run OpenCPN, my plugin does not load. I checked the log and it says:
`Error: path/to/library undefined symbol: g_pBoundaryList`
I also tried using DECL_EXP but still didnt get anything. Is there a way to access this variable through another plugin?

Any help regarding this is appreciated!

Best,
Harshil
